Detectors/TPC/calibration/SpacePoints/src/TrackInterpolation.cxx

```diff
@@ -54,28 +57,171 @@ void TrackInterpolation::init()
   mGeoTRD = o2::trd::Geometry::instance();
   mParams = &SpacePointsCalibConfParam::Instance();
 
+  mSourcesConfigured = src;
+  mTrackTypes.insert({GTrackID::ITSTPC, 0});
+  mTrackTypes.insert({GTrackID::ITSTPCTRD, 1});
+  mTrackTypes.insert({GTrackID::ITSTPCTOF, 2});
+  mTrackTypes.insert({GTrackID::ITSTPCTRDTOF, 3});
+
   mInitDone = true;
-  LOG(info) << "Done initializing TrackInterpolation";
+  LOGP(info, "Done initializing TrackInterpolation. Configured track input: {}", GTrackID::getSourcesNames(mSourcesConfigured));
 }
 
-void TrackInterpolation::process(const o2::globaltracking::RecoContainer& inp, const std::vector<GTrackID>& gids, const std::vector<o2::globaltracking::RecoContainer::GlobalIDSet>& gidTables, std::vector<o2::track::TrackParCov>& seeds, const std::vector<float>& trkTimes, const std::unordered_map<int, int>& trkCounters)
+bool TrackInterpolation::isInputTrackAccepted(GTrackID gid, o2::globaltracking::RecoContainer::GlobalIDSet gidTable, o2::dataformats::PrimaryVertex pv)
 {
-  // main processing function
+  LOGP(debug, "Check if input track {} is accepted", gid.asString());
+  bool hasOuterPoint = gid.getSource() >= GTrackID::ITSTPCTRD; // assuming the ordering of the GID sources will not change in the future
```

Comment:
I think it is better to check explicitly `gidTable[GTrackID::TRD].isIndexSet() || gidTable[GTrackID::TOF].isIndexSet()`, the GTrackID enums may eventually change.

Reply:
also fixed